### PR TITLE
Optimize reserved product quantity update

### PR DESCRIPTION
**Pull request by jocel1**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Optimize the query updating reserved product quantity
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-4843
| How to test?  | Access Catalog>Stock with a lot of historyOrders

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8790)

### CODE REVIEW

File: install-dev/upgrade/sql/1.7.3.1.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+SET SESSION sql_mode = '';
+SET NAMES 'utf8';
+
+ALTER TABLE `PREFIX_order_detail` DROP KEY product_id, ADD KEY product_id (product_id, product_attribute_id);
```
Comment (Quetzacoalt91) [inline install-dev/upgrade/sql/1.7.3.1.sql]: If you're on the develop branch, this should be in the file 1.7.4.0.sql

**Review by Quetzacoalt91 — DISMISSED** [review]

File: install-dev/upgrade/sql/1.7.3.1.sql (hunk 1 above)
Comment (jocel1) [reply install-dev/upgrade/sql/1.7.3.1.sql]: ah, I'll move it in 1.7.3.x branch, thanks!

**Comment by LittleBigDev** [comment]
:lgtm:

---

Reviewed 3 of 5 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/prestashop/prestashop/8790#-:-L5yCPAS2RZ8xNV-QsUa:bnfp4nl)*

**Review by LittleBigDev — DISMISSED** [review]
Thank you @jocel1

**Review by mickaelandrieu — APPROVED** [review]

**Comment by mickaelandrieu** [comment]
Thanks @jocel1 !

**Pull request merged**